### PR TITLE
[29_1] Unit tests for making attachments edge cases

### PR DESCRIPTION
**Pull request by tangdouer1005**

### CODE REVIEW

File: src/Plugins/Pdf/pdf_hummus_extract_attachment.cpp [hunk 1]
```diff
@@ -69,8 +69,14 @@ extract_attachments_from_pdf (url pdf_path, list<url>& names) {
       break;
     }
     unsigned long n= arr->GetLength ();
+    if (n == 0) {
+      if (DEBUG_CONVERT) debug_convert << "n is 0\n";
```
Comment (darcy-shen) [inline src/Plugins/Pdf/pdf_hummus_extract_attachment.cpp:73]: 这里可以明确说明一下n的具体意思

File: src/Plugins/Pdf/pdf_hummus_make_attachment.cpp [hunk 2]
```diff
@@ -63,6 +64,26 @@ class PDFAttachmentWriter : public DocumentContextExtenderAdapter {
 bool
 pdf_hummus_make_attachments (url pdf_path, list<url> attachment_paths,
                              url out_path) {
+  if (N (attachment_paths) == 0) {
+    if (DEBUG_CONVERT) debug_convert << "attachment_paths number is 0\n";
```
Comment (darcy-shen) [inline src/Plugins/Pdf/pdf_hummus_make_attachment.cpp:68]: N (attachment_paths) is 0 可以这样写

**Comment by darcy-shen** [comment]
标题可以改为 Unit tests for edge cases

File: src/Plugins/Pdf/pdf_hummus_extract_attachment.cpp [hunk 3]
```diff
@@ -69,20 +69,29 @@ extract_attachments_from_pdf (url pdf_path, list<url>& names) {
       break;
     }
     unsigned long n= arr->GetLength ();
+    // Every two elements in the array represent an attachment
+    if (n == 0) {
+      if (DEBUG_CONVERT) debug_convert << "arr->GetLength () is 0\n";
+      status= PDFHummus::eFailure;
+      break;
+    }
     if (n & 1) {
```
Comment (darcy-shen) [inline src/Plugins/Pdf/pdf_hummus_extract_attachment.cpp:78]: 后续在lolly里面加一个 number.hpp 然后加一下 is_odd, is_even 这类函数吧
